pkg/apis/extensions/validation/validation.go

```diff
@@ -346,15 +346,20 @@ func ValidateJobSpec(spec *extensions.JobSpec) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldInvalid("completions", spec.Completions, isNegativeErrorMsg))
 	}
 
-	selector := labels.Set(spec.Selector).AsSelector()
-	if selector.Empty() {
-		allErrs = append(allErrs, errs.NewFieldRequired("selector"))
+	if len(spec.Selector.MatchLabels)+len(spec.Selector.MatchExpressions) == 0 {
```

Comment:
Technically, Ii both are empty, the selector matches all pods. The selector can't be nil (after defaulting).